### ISSUE TITLE
Rippled debug crash: "Assertion failed: (continueCallback && !continueCallback())"

### ISSUE THREAD

**Issue by manojsdoshi** [XRPLF/rippled#4745]
## Issue Description
Rippled keeps crashing locally with the following output 

`{
        "Fee" : "20",
        "account" : "raszmy2tJyGsUvzctWjWbiz5u4BcjLjMKA",
        "command" : "account_objects",
        "deletion_blockers_only" : false,
        "ledger_index" : "validated",
        "limit" : 400
}

2023-Oct-04 04:30:40.669786 UTC Resource:DBG Inactive 127.0.0.1:1
2023-Oct-04 04:30:40.669817 UTC Server:DBG Reply: {"result":{"account":"raszmy2tJyGsUvzctWjWbiz5u4BcjLjMKA","account_objects":[],"ledger_hash":"D01780D3CE8E930B3B7EDFCF0F6A27284F6EDA1CC86DEBABBC86D76765676966","ledger_index":417962,"status":"success","validated":true}}

2023-Oct-04 04:30:40.673743 UTC JobQueue:DBG addRefCountedJob : Adding job : RPC-Client : 8
2023-Oct-04 04:30:40.673889 UTC Resource:DBG New unlimited endpoint 127.0.0.1:1
2023-Oct-04 04:30:40.673961 UTC Server:DBG Query: ripple_path_find
{
        "Fee" : "20",
        "destination_account" : "raszmy2tJyGsUvzctWjWbiz5u4BcjLjMKA",
        "destination_amount" : 
        {
                "currency" : "USD",
                "issuer" : "rMUgifyaqshgF3747jbTCeAKRXzqX9PCKp",
                "value" : "10000"
        },
        "source_account" : "rsFCKaQDNDPQiubXUnLeb9Us9ps5npczye",
        "source_currencies" : 
        [
                
                {
                        "currency" : "XRP"
                },
                
                {
                        "currency" : "USD"
                }
        ]
}

2023-Oct-04 04:30:40.674016 UTC PathRequest:DBG 11 created
2023-Oct-04 04:30:40.674041 UTC PathRequest:DBG getLineCache has cache for 0, considering 417962
2023-Oct-04 04:30:40.674062 UTC PathRequest:DBG getLineCache creating new cache for 417962
2023-Oct-04 04:30:40.674083 UTC RippleLineCache:DBG created for ledger 417962
2023-Oct-04 04:30:40.674170 UTC PathRequest:DBG 11 valid: rsFCKaQDNDPQiubXUnLeb9Us9ps5npczye
2023-Oct-04 04:30:40.674199 UTC PathRequest:DBG 11 deliver: 10000/USD/rMUgifyaqshgF3747jbTCeAKRXzqX9PCKp
2023-Oct-04 04:30:40.674223 UTC RippleLineCache:DBG destroyed for ledger 417962 with 1 accounts and 0 distinct trust lines.
2023-Oct-04 04:30:40.674244 UTC LedgerMaster:DBG newPFWork: Creating job. path find threads: 0
2023-Oct-04 04:30:40.674264 UTC JobQueue:DBG addRefCountedJob : Adding job : pf:newRequest : 14
2023-Oct-04 04:30:40.674304 UTC LedgerMaster:DBG updatePaths running
2023-Oct-04 04:30:40.674337 UTC LedgerMaster:DBG Updating paths
2023-Oct-04 04:30:40.674360 UTC PathRequest:DBG getLineCache has cache for 0, considering 417962
2023-Oct-04 04:30:40.674556 UTC PathRequest:DBG getLineCache creating new cache for 417962
2023-Oct-04 04:30:40.674597 UTC RippleLineCache:DBG created for ledger 417962
2023-Oct-04 04:30:40.674622 UTC PathRequest:NFO 11 aborting early
2023-Oct-04 04:30:40.674642 UTC PathRequest:DBG 11 update normal
2023-Oct-04 04:30:40.674698 UTC PathRequest:DBG 11 processing at level 2
2023-Oct-04 04:30:40.674726 UTC PathRequest:DBG 11 Trying to find paths: 1/XRP
2023-Oct-04 04:30:40.674768 UTC Pathfinder:DBG Non-existent gateway
Assertion failed: (continueCallback && !continueCallback()), function findPaths, file PathRequest.cpp, line 541.
`

## Environment
Local running

## Supporting Files

[Uploading debug.log…]()

**Comment by scottschurr** [comment]
The assert you are hitting...
```
Assertion failed: (continueCallback && !continueCallback()), function findPaths, file PathRequest.cpp, line 541.
```
was introduced about a year ago in this commit:
https://github.com/XRPLF/rippled/commit/e7e672c3f832f799c75a93a06540e560951b1278
Perhaps @ximinez remembers why that assert was added and the purpose it serves?

**Comment by ximinez** [comment]
Duplicate of #4744

**Comment by intelliot** [comment]
Fixed by #4743 and expected to be released with 2.0.0-b3

**Comment by ncirelli87** [comment]
I'll call you back WXRP style
